### ISSUE TITLE
Problem with replaceAll

### ISSUE THREAD

**Issue by dawnerd** [meilisearch/instant-meilisearch#15]
Hey all

I know this is early in the modules life but I stumbled on this and it's exactly what I was about to build out. Gave it a go and it wasn't rendering the hits so I did some debugging and it looks like these lines https://github.com/meilisearch/instant-meilisearch/blob/master/src/index.js#L36-L37 are silently failing.

As a stop gap I had to use https://github.com/es-shims/String.prototype.replaceAll

Looks like your demo app has the same problem so hope this helps :)

Edit: This was with react fyi

**Comment by curquiza** [comment]
Hello @dawnerd!

That is exactly the kind of feedback I want! Thanks for your issue, especially because if we are in an early stage 🙂

> Looks like your demo app has the same problem so hope this helps :)

What do you mean? Do you mean this [demo app](https://codesandbox.io/s/ms-react-is-sh9ud?fontsize=14&hidenavigation=1&theme=dark&file=/src/App.js)?
Because I don't succeed to reproduce your error.

Can you show me how you use `instant-meilisearch` and InstantSearch so that I can reproduce it? 🙂

**Comment by dawnerd** [comment]
You can see in the screenshot below, the request is fired off and results are returned but nothing is rendered. and no errors in console.

I'm in Chrome Version 84.0.4147.105 (Official Build) (64-bit)

Looks like replaceAll was only just recently added to Chrome 85 (https://caniuse.com/#feat=mdn-javascript_builtins_string_replaceall) which isn't out yet.

Although I think the bigger issue is for whatever reason that error silently failing.

This is what I did for debugging locally:

```js
try {
        parsedResponse = {
          index: indexUid,
          hitsPerPage: this.hitsPerPage,
          facets,
          exhaustiveFacetsCount,
          exhaustiveNbHits,
          nbHits,
          processingTimeMs,
          query,
          ...this.paginationParams(hits.length, params),
          hits: this.parseHits(hits, params), // Apply pagination + highlight
        }
      } catch (e) {
        console.error(e);
      }
```

![image](https://user-images.githubusercontent.com/24781/89127652-fbf33800-d4a3-11ea-9266-1e98d816fb0a.png)

**Comment by dawnerd** [comment]
Found another minor issue with this same section of code. If there's a result that has a field set to null, `result[key] = { value: newHighlightString.toString() }` will fail - again silently.

A quick fix I did was to change `let newHighlightString = formattedHit[key]` to `let newHighlightString = formattedHit[key] || ''`.

**Comment by curquiza** [comment]
> I'm in Chrome Version 84.0.4147.105 (Official Build) (64-bit)

Oh! Indeed now I can reproduce the issue 🙂

> Although I think the bigger issue is for whatever reason that error silently failing.

This is indeed the main problem...

> Found another minor issue with this same section of code. If there's a result that has a field set to null, `result[key] = { value: newHighlightString.toString() }` will fail - again silently.
> 
> A quick fix I did was to change `let newHighlightString = formattedHit[key]` to `let newHighlightString = formattedHit[key] || ''`.

Thanks for this fix! I will update that soon


In general, thanks for your bug reports! It helps!
We are going to investigate on all these points!

**Comment by curquiza** [comment]
Partially closed with #22

**Comment by curquiza** [comment]
Totally fixed with #25 
New issue opened about silent errors with #27
